GoogleApis.Blazor/Calendar/CalendarService.cs

```diff
@@ -108,21 +104,21 @@ public string GetCalendars(int maxResults = 250, bool forceAccessToken = false)
         /// <param name="calendarId"></param>
         /// <param name="forceAccessToken">If true and access token expired, it automatically calls for new access token with refresh token.</param>
         /// <returns></returns>
-        public string GetCalendarById(string calendarId, bool forceAccessToken = false)
+        public async Task<GoogleCalendarListModel> GetCalendarById(string calendarId, bool forceAccessToken = false)
         {
             var client = HttpClientFactory.CreateClient();
-            var result = client.GetAsync($"https://www.googleapis.com/calendar/v3/users/me/calendarList/{calendarId}?access_token=" + _accessToken).Result;
+            var result = await client.GetAsync($"https://www.googleapis.com/calendar/v3/users/me/calendarList/{calendarId}?access_token=" + _accessToken);
 
-            string contentResult = result.Content.ReadAsStringAsync().Result;
+            string contentResult = await result.Content.ReadAsStringAsync();
+            var model = JsonSerializer.Deserialize<GoogleCalendarListModel>(contentResult);
 
             if (forceAccessToken == false || !AuthService.IsAccessTokenExpired(contentResult))
             {
-                return contentResult;
+                return model;
             }
 
-            AccessToken = AuthService.RefreshAccessToken(_refreshToken);
-            GetCalendarById(calendarId);
-            return "";
+            AccessToken = await AuthService.RefreshAccessToken(_refreshToken);
+            return await GetCalendarById(calendarId);
```

Comment:
Its not about your PR i know, but doesn't this line may cause infinite loop? I think we should call the method again only if AccessToken is not null and different from the previous (the first access token we not refreshed with refresh token) one?

Reply:
@mckaragoz I've resolved these, thanks.